### PR TITLE
Make view converters operate on samples only

### PR DESCRIPTION
**Pull request by klecki**
## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [x] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
`view` for TensorList now accesses samples only
instead of the underlying contiguous buffer.
Remove `view_as_tensor` - never used for TensorList
other than in tests, for Tensor just `view` can
be used.

#### Additional information
- Affected modules and functionalities:
`view`

- Key points relevant for the review:

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2255

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

File: dali/pipeline/data/views.h [hunk 1]
```diff
@@ -74,14 +75,21 @@ TensorShape<ndim> get_tensor_shape(const TensorList<Backend> &tl) {
   return convert_dim<ndim>(out);
 }
 
+
```
Comment (mzient) [inline dali/pipeline/data/views.h:78]: This function does almost the same thing as the one with TensorVector, but it's written differently - I think these function should be as similar as they can be - and placed next to each other.

Comment (klecki) [reply dali/pipeline/data/views.h:78]: done

**Comment by klecki** [comment]
!build

File: dali/pipeline/data/views.h [hunk 2]
```diff
@@ -181,16 +164,17 @@ view(const TensorVector<Backend> &data) {
...
-    ret.data[i] = data[i].template data<U>();
+  const auto &shape = data.shape();
+  detail::enforce_dim_in_view<ndim>(shape);
+
+  std::vector<T *> ptrs(shape.num_samples());
+  for (int i = 0; i < shape.num_samples(); i++) {
+    ptrs[i] = data[i].template data<U>();
   }
-  return ret;
+  return { std::move(ptrs), convert_dim<ndim>(shape) };
```
Comment (mzient) [inline dali/pipeline/data/views.h:174]: Out of sheer curiosity, I've checked - this variant is over 10% faster. Good job!
:+1:

Comment (klecki) [reply dali/pipeline/data/views.h:174]: It might be due to the fact that we do not copy the shape twice?

**Review by stiepan — APPROVED** [review]

**Review by mzient — APPROVED** [review]

**Pull request merged**